### PR TITLE
Add workflow start exit endpoints

### PR DESCRIPTION
**Pull request by olaals**

### CODE REVIEW

File: api/Controllers/WorkflowsControlller.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+public class WorkflowExitedNotification
+{
+    public required string InspectionId { get; set; }
+    public required string WorkflowStatus { get; set; }
+}
+
+[ApiController]
+[Route("[controller]")]
+public class WorkflowsController(ILogger<InspectionDataController> logger, IInspectionDataService inspectionDataService) : ControllerBase
```
Comment (Christdej) [inline api/Controllers/WorkflowsControlller.cs:21]: Can remove logger as it is unused, or we can use it

File: api/Controllers/WorkflowsControlller.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    [HttpPut]
+    [Route("notify-workflow-started")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<InspectionDataResponse>> WorkflowStarted([FromBody] WorkflowStartedNotification notification)
+    {
+        var updatedInspectionData = await inspectionDataService.UpdateAnonymizerWorkflowStatus(notification.InspectionId, WorkflowStatus.Started);
+        if (updatedInspectionData == null)
+        {
+            return NotFound();
```
Comment (Christdej) [inline api/Controllers/WorkflowsControlller.cs:33]:
```suggestion
            return NotFound($"Could not find workflow with inspection id {inspectionId}");
```

File: api/Controllers/WorkflowsControlller.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        else
+        {
+            status = WorkflowStatus.ExitFailure;
+        }
+
+        var updatedInspectionData = await inspectionDataService.UpdateAnonymizerWorkflowStatus(notification.InspectionId, status);
+        if (updatedInspectionData == null)
+        {
+            return NotFound();
```
Comment (Christdej) [inline api/Controllers/WorkflowsControlller.cs:59]:
```suggestion
            return NotFound($"Could not find workflow with inspection id {inspectionId}");
```

**Review by Christdej — CHANGES_REQUESTED** [review]

File: api/Controllers/WorkflowsControlller.cs (hunk 1 above)
Comment (olaals) [reply api/Controllers/WorkflowsControlller.cs]: Removed it for now

File: api/Controllers/WorkflowsControlller.cs (hunk 2 above)
Comment (olaals) [reply api/Controllers/WorkflowsControlller.cs]: Accepted suggestion

**Review by Christdej — APPROVED** [review]
LGTM

**Pull request merged**